src/raiden_libs/blockchain.py

```diff
@@ -218,19 +222,21 @@ def get_monitoring_blockchain_events(
         if event_name == MonitoringServiceEvent.NEW_BALANCE_PROOF_RECEIVED:
             events.append(
                 ReceiveMonitoringNewBalanceProofEvent(
-                    token_network_address=decode_hex(event["args"]["token_network_address"]),
+                    token_network_address=TokenNetworkAddress(
+                        decode_hex(event["args"]["token_network_address"])
```

Comment:
Again, to_canonical_address would be more consistent then using decode_hex which is just possible due to the typecast.